linkcheck/checker/internpaturl.py

```diff
@@ -16,15 +16,18 @@
 """
 Intern URL pattern support.
 """
+
 import re
+import urllib.parse
+
 from . import urlbase, absolute_url
-from .. import strformat, url as urlutil
+from .. import url as urlutil
 
 
 def get_intern_pattern(url):
     """Return intern pattern for given URL. Redirections to the same
     domain with or without "www." prepended are allowed."""
-    parts = strformat.url_unicode_split(url)
+    parts = list(urllib.parse.urlsplit(url))
```

Comment:
No need for list(), it's a named tuple that supports indexing.

(But for readability we ought to refer to its members by name.  Separate PR material.)

Reply:
Removed list()